internal/storage/bucket_handle_test.go

```diff
@@ -467,3 +469,94 @@ func (t *BucketHandleTest) TestUpdateObjectMethodWithMissingObject() {
 
 	AssertTrue(errors.As(err, &notfound))
 }
+
+func (t *BucketHandleTest) TestComposeObjectMethodWithValidObject() {
+	obj, err := t.bucketHandle.StatObject(context.Background(),
+		&gcs.StatObjectRequest{
+			Name: TestObjectName,
+		})
+
+	AssertEq(nil, err)
+
+	composedObj, err := t.bucketHandle.ComposeObjects(context.Background(),
+		&gcs.ComposeObjectsRequest{
+			DstName:                       dstObjectName,
+			DstGenerationPrecondition:     nil,
+			DstMetaGenerationPrecondition: nil,
+			Sources: []gcs.ComposeSource{
+				{
+					Name: TestObjectName,
+				},
+				{
+					Name: TestSubObjectName,
+				},
+			},
+			ContentType:     ContentType,
+			ContentEncoding: ContentEncoding,
+			ContentLanguage: ContentLanguage,
+			CacheControl:    CacheControl,
+			CustomTime:      CustomTime,
+			EventBasedHold:  true,
+			StorageClass:    StorageClass,
+			Metadata: map[string]string{
+				MetaDataKey: MetaDataValue,
+			},
+			Acl: nil,
+		})
+
+	AssertEq(nil, err)
+	AssertNe(nil, obj)
+	AssertGe(composedObj.Size, obj.Size)
+}
+
+func (t *BucketHandleTest) TestComposeObjectMethodWithSrcObjectDoesNotExist() {
+	_, err := t.bucketHandle.ComposeObjects(context.Background(),
+		&gcs.ComposeObjectsRequest{
+			DstName:                       TestObjectName,
+			DstGenerationPrecondition:     nil,
+			DstMetaGenerationPrecondition: nil,
+			Sources: []gcs.ComposeSource{
+				{
+					Name: missingObjectName,
+				},
+			},
+			ContentType:     ContentType,
+			ContentEncoding: ContentEncoding,
+			ContentLanguage: ContentLanguage,
+			CacheControl:    CacheControl,
+			CustomTime:      CustomTime,
+			EventBasedHold:  true,
+			StorageClass:    StorageClass,
+			Metadata: map[string]string{
+				MetaDataKey: MetaDataValue,
+			},
+			Acl: nil,
+		})
+
+	// For fakeobject it is giving googleapi 500 error, where as in real mounting we are getting "404 not found error"
+	AssertNe(nil, err)
+}
+
+func (t *BucketHandleTest) TestComposeObjectMethodWithInNilSources() {
+	_, err := t.bucketHandle.ComposeObjects(context.Background(),
+		&gcs.ComposeObjectsRequest{
+			DstName:                       TestObjectName,
+			DstGenerationPrecondition:     nil,
+			DstMetaGenerationPrecondition: nil,
+			Sources:                       nil,
+			ContentType:                   ContentType,
+			ContentEncoding:               ContentEncoding,
+			ContentLanguage:               ContentLanguage,
+			CacheControl:                  CacheControl,
+			CustomTime:                    CustomTime,
+			EventBasedHold:                true,
+			StorageClass:                  StorageClass,
+			Metadata: map[string]string{
+				MetaDataKey: MetaDataValue,
+			},
+			Acl: nil,
+		})
+
+	// error : Error in composing object: storage: at least one source object must be specified
```

Comment:
is it not possible to compare the error type?

Reply:
error type is default. how should I compare. I didn't get
https://github.com/GoogleCloudPlatform/gcsfuse/blob/0a63e134268dc8cc81f5f9d5d7e1b43feb3779db/internal/storage/bucket_handle.go#L347